vm_method.c

```diff
@@ -771,7 +763,7 @@ make_method_entry_refined(VALUE owner, rb_method_entry_t *me)
             rb_method_entry_alloc(me->called_id, me->owner,
                                   me->defined_class ?
                                   me->defined_class : owner,
-                                  method_definition_addref(me->def));
+                                  me->def);
```

Comment:
I think we need to increment the reference count here.

Reply:
If I increment here, the leak comes back. I agree it makes sense that this would increment the count, but without this increment we do seem to clean up correctly. My only theory is it is still the correct count because `reference_count` starts at 1, so it still counts itself correctly?

Reply:
Peter helped me find `rb_method_definition_set` can correct this count when it goes wrong. Thanks! ❤️